src/transformers/models/auto/modeling_auto.py

```diff
@@ -1671,7 +1671,14 @@ def _prepare_config_for_auto_class(cls, config: PretrainedConfig) -> PretrainedC
         Under the hood, multimodal models mapped by AutoModelForCausalLM assume the text decoder receives its own
         config, rather than the config for the whole model. This is used e.g. to load the text-only part of a VLM.
         """
-        return config.get_text_config(decoder=True)
+        text_config, text_config_names = config.get_text_config(decoder=True, return_text_config_name=True)
```

Comment:
Can we have some custom code here instead of adding custom logic to `get_text_config`? 🙏 

In its current form, the output interface of `get_text_config` becomes much more complex (`"PretrainedConfig"` to `Union["PretrainedConfig", Tuple["PretrainedConfig", List[str]]]`). `get_text_config` is used in many places, but the only place where the extra logic is needed is here -> easier to maintain `get_text_config`